build/storage/tests/it/ptf_tests/tests/test_hot_plug.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+from ptf.base_tests import BaseTest
+from system_tools.test_platform import (HostTargetPlatform, IPUStoragePlatform,
+                                        StorageTargetPlatform)
+
+from tests.steps.ramdrive import (
+    CreateRamdriveAndAttachAsNsToSubsystem64Step,
+    CreateRamdriveAndAttachAsNsToSubsystemAbove64Step,
+    CreateRamdriveAndAttachAsNsToSubsystemStep)
+from tests.steps.subsystem import CreateAndExposeSubsystemOverTCPStep
+from tests.steps.virtio_blk import (CreateVirtioBlk64Step,
+                                    CreateVirtioBlkAbove64Step,
+                                    CreateVirtioBlkStep, DeleteVirtioBlk64Step,
+                                    DeleteVirtioBlkStep)
+
+
+class Test1HotPlug(BaseTest):
+    def setUp(self):
+        self.storage_target_platform = StorageTargetPlatform()
+        self.ipu_storage_platform = IPUStoragePlatform()
+        self.host_target_platform = HostTargetPlatform()
+
+    def runTest(self):
+        CreateAndExposeSubsystemOverTCPStep(self.storage_target_platform).run()
+        volume_ids = (
+            CreateRamdriveAndAttachAsNsToSubsystemStep(self.storage_target_platform)
+            .run()
+            .get("volume_ids")
+        )
+        device_handles = (
+            CreateVirtioBlkStep(
+                self.host_target_platform,
+                self.ipu_storage_platform,
+                volume_ids,
+            )
+            .run()
+            .get("device_handles")
+        )
+        DeleteVirtioBlkStep(
+            self.host_target_platform,
+            self.ipu_storage_platform,
+            device_handles,
+        ).run()
+
+    def tearDown(self):
+        self.storage_target_platform.clean()
+        self.host_target_platform.clean()
+        self.ipu_storage_platform.clean()
+
+
+class Test64HotPlug(BaseTest):
+    def setUp(self):
+        self.storage_target_platform = StorageTargetPlatform()
+        self.ipu_storage_platform = IPUStoragePlatform()
+        self.host_target_platform = HostTargetPlatform()
+
+    def runTest(self):
+        CreateAndExposeSubsystemOverTCPStep(self.storage_target_platform).run()
+        volume_ids = (
+            CreateRamdriveAndAttachAsNsToSubsystem64Step(self.storage_target_platform)
```

Comment:
I am still thinking if we have transform the tests to something more linear and easier to comprehend.
Will this variant be better?
```
self.assertFalse(self.storage_target_platform.is_port_used(config.nvme_controller_port))
self.storage_target_platform.create_subsystem()
self.assertTrue(self.storage_target_platform.is_port_used(config.nvme_controller_port))

volume_ids=self.storage_target_platform.create_ramdrives(config.max_number_of_virtio_blks)
self.assertEqual(len(volume_ids), config.max_number_of_virtio_blks)
# check elements in volume_ids are not empty and valid uuids

self.assertEqual(self.host_target_platform.get_number_of_virtio_blk_devices(), 0)
device_handles=self.ipu_platform.create_virtio_blk_device(volume_ids)
self.assertEqual(self.host_target_platform.get_number_of_virtio_blk_devices(), len(volume_ids))

self.ipu_platform.delete_virtio_blk_devices(device_handles)
self.assertEqual(self.host_target_platform.get_number_of_virtio_blk_devices(), 0)
```

I remember that you are concerned about checking as many pre/post conditions as possible to identify the exact step we failed.
What if we group those calls into functions? Will this option fit our need then?
```
# in self.create_subsystem_and_verify(...)
self.assertFalse(storage_target_platform.is_port_used(nvme_controller_port))
storage_target_platform.create_subsystem()
self.assertTrue(storage_target_platform.is_port_used(nvme_controller_port))

# in self.create_ramdrives_and_verify(...)
volume_ids=storage_target_platform.create_ramdrives(max_number_of_virtio_blks)
self.assertEqual(len(volume_ids), max_number_of_virtio_blks)
# check elements in volume_ids are not empty and valid uuids

# in self.create_virtio_blks_and_verify(...)
self.assertEqual(host_target_platform.get_number_of_virtio_blk_devices(), 0)
device_handles=ipu_platform.create_virtio_blk_device(volume_ids)
self.assertEqual(host_target_platform.get_number_of_virtio_blk_devices(), len(volume_ids))

# in delete_virtio_blks_and_verify(...)
ipu_platform.delete_virtio_blk_devices(device_handles)
self.assertEqual(host_target_platform.get_number_of_virtio_blk_devices(), 0)
```

As one additional option we could also try to simplify the existing classes
```
volume_ids=CreateRamdriveAndAttachAsNsToSubsystemNStep(self.storage_target_platform, config.max_supported_virtio_blks)()
```
But I am not sure if this is going to be simpler than a simple function call.
@intelfisz what do you think?

Reply:
Discussed offline and decided to leave it as is. Later we can extract the common part when we have more tests